### PR TITLE
Remove ecma_simple_value_t and refactor ecma_make_simple_value

### PR DESCRIPTION
**Pull request by rerobika**
This patch removes all ecma_make_simple_value calls to make the code more easy to understand.
Also removes the type ecma_simple_value_t which improves the performance in related code paths by calculating the value of new ecma_value_t is no longer needed.

JerryScript-DCO-1.0-Signed-off-by: Robert Fancsik frobert@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/ecma/base/ecma-globals.h [hunk 1]
```diff
@@ -151,6 +128,35 @@ typedef int32_t ecma_integer_value_t;
  */
 #define ECMA_DIRECT_SHIFT 4
 
+/* ECMA make simple value */
+#define ECMA_MAKE_SIMPLE_VALUE(simple_value) \
+(((((ecma_value_t) (simple_value)) << ECMA_DIRECT_SHIFT) | ECMA_DIRECT_TYPE_SIMPLE_VALUE))
```
Comment (zherczeg) [inline jerry-core/ecma/base/ecma-globals.h]: Two spaces before definition.

File: jerry-core/ecma/base/ecma-helpers-value.c [hunk 2]
```diff
@@ -52,10 +52,6 @@ JERRY_STATIC_ASSERT (sizeof (uintptr_t) > sizeof (ecma_value_t),
 
 #endif /* ECMA_VALUE_CAN_STORE_UINTPTR_VALUE_DIRECTLY */
 
-JERRY_STATIC_ASSERT ((ECMA_SIMPLE_VALUE_FALSE | 0x1) == ECMA_SIMPLE_VALUE_TRUE
-                     && ECMA_SIMPLE_VALUE_FALSE != ECMA_SIMPLE_VALUE_TRUE,
-                     only_the_lowest_bit_must_be_different_for_simple_value_true_and_false);
```
Comment (zherczeg) [inline jerry-core/ecma/base/ecma-helpers-value.c]: This assert must be kept just rewrite it to `(1 << ECMA_DIRECT_SHIFT)`

File: jerry-core/vm/opcodes-ecma-bitwise.c [hunk 3]
```diff
@@ -45,7 +45,48 @@ do_number_bitwise_logic (number_bitwise_logic_op op, /**< number bitwise logic o
   JERRY_ASSERT (!ECMA_IS_VALUE_ERROR (left_value)
                 && !ECMA_IS_VALUE_ERROR (right_value));
 
-  ecma_value_t ret_value = ecma_make_simple_value (ECMA_SIMPLE_VALUE_EMPTY);
+  if (ecma_are_values_integer_numbers (left_value, right_value))
```
Comment (zherczeg) [inline jerry-core/vm/opcodes-ecma-bitwise.c]: What is the reason for this change? Seems unrelated.

**Comment by rerobika** [comment]
@zherczeg I've corrected all your remarks.

File: jerry-core/ecma/base/ecma-globals.h [hunk 4]
```diff
@@ -151,6 +128,35 @@ typedef int32_t ecma_integer_value_t;
...
+   *   - special register or stack values for vm
+   */
+  ECMA_SIMPLE_VALUE_EMPTY = ECMA_MAKE_SIMPLE_VALUE (0), /**< uninitialized value */
+  ECMA_SIMPLE_VALUE_ERROR = ECMA_MAKE_SIMPLE_VALUE (1), /**< an error is currently thrown */
+  ECMA_SIMPLE_VALUE_FALSE = ECMA_MAKE_SIMPLE_VALUE (2), /**< boolean false */
+  ECMA_SIMPLE_VALUE_TRUE = ECMA_MAKE_SIMPLE_VALUE (3), /**< boolean true */
+  ECMA_SIMPLE_VALUE_UNDEFINED = ECMA_MAKE_SIMPLE_VALUE (4), /**< undefined value */
+  ECMA_SIMPLE_VALUE_NULL = ECMA_MAKE_SIMPLE_VALUE (5), /**< null value */
+  ECMA_SIMPLE_VALUE_ARRAY_HOLE = ECMA_MAKE_SIMPLE_VALUE (6), /**< array hole, used for
+                                                              * initialization of an array literal */
```
Comment (zherczeg) [inline jerry-core/ecma/base/ecma-globals.h]: Align the two comment lines below each other.
```
 /**< text text text
  *   text text text */
```

Comment (zherczeg) [reply jerry-core/ecma/base/ecma-globals.h]: (For each comment)

**Comment by dbatyai** [comment]
@rerobika, Please rebase to master.

**Comment by glistening** [comment]
What about using `ECMA_VALUE_EMPTY`, `ECMA_VALUE_ERROR`, `ECMA_VALUE_FALSE`, ...  instead of `ECMA_SIMPLE_VALUE_EMPTY`, `ECMA_SIMPLE_VALUE_ERROR`, ...? `ecma_simple_value_t` type is removed. We don't need to expose this name on exposed name any more.

**Comment by rerobika** [comment]
@glistening That's a good idea and I like it. @zherczeg @dbatyai What's your opinion about this renaming suggestion?

**Comment by zherczeg** [comment]
Good idea
